fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/functions/scalar/ArrayMap.java

```diff
@@ -66,6 +66,13 @@ public ArrayMap withChildren(List<Expression> children) {
     public DataType getDataType() {
         Preconditions.checkArgument(children.get(0) instanceof Lambda,
                 "The first arg of array_map must be lambda");
+        Preconditions.checkArgument(children.size() > 1,
+                "The args of array_map must more than 1");
+        // children after lambda must be array
+        for (int i = 1; i < children.size(); i++) {
+            Preconditions.checkArgument(children.get(i).getDataType() instanceof ArrayType,
+                    "The next arg of array_map must be array");
```

Comment:
what does "The next arg" mean?

Reply:
here is no need to add this check , because child of lambda is ArrayItemReference, which already has type limit